Onetrust doc revamp (#1142)
rudderlabs/rudderstack-docs: docs/sources/event-streams/sdks/rudderstack-javascript-sdk/onetrust-consent-manager.mdx

The JavaScript SDK seamlessly integrates with the OneTrust SDK. It lets you map the OneTrust cookie/consent groups to RudderStack's consent purposes. RudderStack, in turn, uses this consent information to enable/disable tracking and sending the data.

<div class="warningBlock">

> The OneTrust integration with the JavaScript SDK is applicable only for the <Link to="/destinations/rudderstack-connection-modes/#device-mode">device mode</Link> connections. For the <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link> connections, you can write a <Link to="/features/transformations/#adding-a-transformation">transformation</Link> to achieve your desired use-case.

Reviewer: ```suggestion
The OneTrust integration with the JavaScript SDK is applicable only for the <Link to="/destinations/rudderstack-connection-modes/#device-mode">device mode</Link> connections. For the <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link> connections, you can write a <Link to="/features/transformations/#adding-a-transformation">transformation</Link> to implement your desired use-case.
```

Reviewer: we can decide if we're going with `use case` or `use-case` when we finalize our style guide.

<!-- OneTrust Cookies Consent Notice end for samplewebsite.com -->
```

<div class="warningBlock">

> As the above snippet shows, you <strong>must</strong> first load the OneTrust script and if the user provides consent, only then load the JavaScript SDK. This is because the JavaScript SDK determines whether to send events to a destination or not based on the user's consent. If a user denies OneTrust, then Rudderstack neither loads the destination SDKs nor sends any events to them.

Reviewer: ```suggestion
As seen in the above snippet, you <strong>must</strong> first load the OneTrust script and then load the JavaScript SDK <strong>only if</strong> the user provides their consent. This is because the SDK determines whether to send events to a destination based on the user's consent. If a user denies consent in OneTrust, then Rudderstack does not load the destination SDKs or send any events to them.
```

Replace <Link to="/resources/glossary/#write-key"><code class="inline-code">WRITE_KEY</code></Link> and <Link to="/dashboard-guides/overview/#data-plane-url"><code class="inline-code">DATA_PLANE_URL</code></Link> in the above snippet with the actual values.
</div>

## Consent management with other managers

If you are using a consent manager other than OneTrust, configure the cookie categories you want to show on your website whenever a new user visits your website.

Next, you need to create an internal mapping between these cookie categories and the destinations configured in RudderStack. For example:
- Analytical cookie: Amplitude, Google Analytics
- Targeting cookie: Bing Ads, Google Ads
- Marketing cookie: Braze, Mailchimp, Customer.io

> When a user provides consent, you should fetch the consent (via API/SDK provided by your consent manager) and filter the destinations depending on the consent. Once you have the list of destinations that the user has provided consent for, you can load the Rudderstack JavaScript SDK for those destinations.

Reviewer: ```suggestion
When a user provides consent, you should fetch the consent (via the API or SDK provided by your consent manager) and filter the destinations depending on the consent. Once you have the list of destinations for which the user has provided consent, you can only load the JavaScript SDK for those destinations.
```

> For example, if a user has provided consent for analytical cookie, then you can load the JavaScript SDK as shown:

Reviewer: ```suggestion
For example, if a user has provided consent for an analytical cookie, then you can load the JavaScript SDK as shown:
```

```javascript
rudderanalytics.load(WRITE_KEY,DATA_PLANE_URL,{
   integrations: {
       All: false,
       "Amplitude": true,
       "Google Analytics": true
       // only provide the destinations the user has provide consent for
   }
   // pass other initialization options
});
```

## FAQ

### Can I integrate the RudderStack JavaScript SDK with any consent manager?

Currently, RudderStack only supports **OneTrust** cookie consent manager. For using other consent managers, you can refer to the <Link to="#consent-management-with-other-managers">Consent management with other managers</Link> section. 

> ### What will happen if RudderStack JavaScript SDK is loaded before the OneTrust modal is closed?

Reviewer: ```suggestion
### What happens if the JavaScript SDK is loaded before the OneTrust modal is closed?
```

> In such a case, the JavaScript SDK will not be able to capture the user consent. Hence, it will load all the connected destinations and all the events will flow through them.

Reviewer: ```suggestion
In this case, the JavaScript SDK will not be able to capture the user consent. Hence, it will load all the connected destinations and all the events will flow through to them.
```
